src/detect-engine-file.c

```diff
@@ -194,7 +194,15 @@ uint8_t DetectFileInspectGeneric(DetectEngineCtx *de_ctx, DetectEngineThreadCtx
     if (ffc == NULL) {
         SCReturnInt(DETECT_ENGINE_INSPECT_SIG_CANT_MATCH_FILES);
     } else if (ffc->head == NULL) {
-        SCReturnInt(DETECT_ENGINE_INSPECT_SIG_NO_MATCH);
+        if (s->flags & SIG_FLAG_FILESTORE) {
```

Comment:
Will this ignore the `file.data` part in 
`alert http any any -> any any (http.uri; content: "toto"; filestore; file.data; content: "tata";)` and match for GET /toto when it should match only for `POST /toto ... tata`

Reply:
Really interesting point to check. I'm going to cook suricata-verify test for that and correct current code if necessary.

Reply:
test added and it was working fine.